test/unit/nics/bmc_test.rb

```diff
@@ -5,4 +5,39 @@ class BMCTest < ActiveSupport::TestCase
     host = FactoryGirl.build(:host, :managed)
     assert FactoryGirl.build(:nic_bmc, :host => host, :provider => 'ipmi').valid?
   end
+
+  context "no BMC smart proxy exists" do
+    def setup
+      SmartProxy.stubs(:with_features).with('BMC').returns([])
+    end
+
+    test 'requires BMC proxy without subnet' do
+      host = FactoryGirl.build(:host, :managed)
+      bmc_nic = FactoryGirl.build(:nic_bmc, :host => host)
+      host.interfaces << bmc_nic
+      refute host.valid?
```

Comment:
Ditto on other tests about validity, try to test just the NIC object rather than all of host.  Even better if you can test that this particular error has been added via assert_valid/refute_with_errors or something.